extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -482,11 +485,110 @@ class LifecycleTask extends BackbeatTask {
                     store[aimu][dai] = rule[aimu][dai];
                 }
             }
+            const hasTransitions =
+                Array.isArray(rule.Transitions) && rule.Transitions.length > 0;
+            if (hasTransitions && enabledRules.includes('transitions')) {
+                const transition = this._getApplicableTransition({
+                    transitions: rule.Transitions,
+                    lastModified: metadata.LastModified,
+                    currentDate,
+                });
+                store.Transition = this._getStoreTransition({
+                    transition1: transition,
+                    transition2: store.Transition,
+                    lastModified: metadata.LastModified,
+                });
+            }
+            // TODO: Add support for NoncurrentVersionTransitions.
             return store;
         }, {});
+        // Do not transition to a location where the object is already stored.
+        if (applicableRules.Transition &&
```

Comment:
I think this check would be better placed inside `getApplicableTransition`, it's generally better if all related criteria to select a rule can be done in a single place to ease understanding why a rule is applicable or not.